spring-cloud-task-docs/src/main/asciidoc/features.adoc

```diff
@@ -233,6 +233,10 @@ custom implementation of the `TaskConfigurer` interface. Typically, extending th
 overriding the required getter is sufficient. However, implementing your own from scratch
 may be required.
 
+NOTE: Users should not directly use getter methods from a TaskConfigurer directly
+unless they are using it to supply beans in a class annotated with
+https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/context/annotation/Configuration.html[Configuration].
```

Comment:
I'll tweak this on merge since technically you could use XML configuration as well...